actors/miner/tests/util.rs

```diff
@@ -741,6 +742,109 @@ impl ActorHarness {
+        for (i, precommit) in precommits.iter().enumerate() {
+            let sector_data = SectorDataSpec {
+                deal_ids: precommit.info.deal_ids.clone(),
+                sector_type: precommit.info.seal_proof,
+            };
+            cdc_inputs.push(sector_data);
+            let comm_d = make_piece_cid(format!("commd-{}", i).as_bytes());
+            comm_ds.push(comm_d);
+        }
+        let cdc_params = ComputeDataCommitmentParams { inputs: cdc_inputs };
+        let cdc_ret = ComputeDataCommitmentReturn { commds: comm_ds.clone() };
+        runtime.expect_send(
+            *STORAGE_MARKET_ACTOR_ADDR,
+            MarketMethod::ComputeDataCommitment as u64,
+            RawBytes::serialize(cdc_params).unwrap(),
+            BigInt::zero(),
+            RawBytes::serialize(cdc_ret).unwrap(),
+            ExitCode::OK,
+        );
+        self.expect_query_network_info(runtime);
+
+        // expect randomness queries for provided precommits
+        let mut seal_rands = Vec::new();
+        let mut seal_int_rands = Vec::new();
+
+        for precommit in precommits.iter() {
+            let seal_rand = Randomness(vec![1, 2, 3, 4]);
+            seal_rands.push(seal_rand.clone());
+            let seal_int_rand = Randomness(vec![5, 6, 7, 8]);
+            seal_int_rands.push(seal_int_rand.clone());
+            let interactive_epoch =
+                precommit.pre_commit_epoch + runtime.policy.pre_commit_challenge_delay;
+
+            let receiver = runtime.receiver;
+            let buf = receiver.marshal_cbor().unwrap();
+            runtime.expect_get_randomness_from_tickets(
+                DomainSeparationTag::SealRandomness,
+                precommit.info.seal_rand_epoch,
+                buf.clone(),
+                seal_rand,
+            );
+            runtime.expect_get_randomness_from_beacon(
+                DomainSeparationTag::InteractiveSealChallengeSeed,
+                interactive_epoch,
+                buf,
+                seal_int_rand,
+            );
+        }
+
+        // verify syscall
+        let mut svis = Vec::new();
+        for (i, precommit) in precommits.iter().enumerate() {
+            svis.push(AggregateSealVerifyInfo {
+                sector_number: precommit.info.sector_number,
+                randomness: seal_rands.get(i).cloned().unwrap(),
+                interactive_randomness: seal_int_rands.get(i).cloned().unwrap(),
+                sealed_cid: precommit.info.sealed_cid,
+                unsealed_cid: comm_ds[i],
+            })
+        }
+        let actor_id = runtime.miner.id().unwrap();
+        runtime.expect_aggregate_verify_seals(svis, params.aggregate_proof.clone(), Ok(()));
+
+        // confirm sector proofs valid
+        self.confirm_sector_proofs_valid_internal(runtime, config, &precommits);
+
+        // burn network fee
+        let expected_fee = aggregate_prove_commit_network_fee(precommits.len() as i64, &base_fee);
+        runtime.expect_send(
```

Comment:
Which edge case exactly? The `expect_send`? The go tests do have this in `minter_test.go` line 4150

Reply:
The edge case missing is when the network fee is 0.  The way to handle this is check that the expected_fee is not 0 before expecting the send.  The go test makes the same mistake, this would be a strict improvement.  No tests yet hit this case (otherwise it would have been fixed in go) but its possible that this helper method gets used in the future with a zero base fee and then this will need to be fixed.

Reply:
Ok, I've just added a simple assert! to check that the expected_fee is greater than 0 right before the expect send. Let me know if this is the right call